cloud/docker/docker.py

```diff
@@ -796,13 +804,22 @@ def get_differing_containers(self):
                     continue
 
             # EXPOSED PORTS
-            # Note that ports that are bound at container run are also exposed
-            # implicitly.
-            expected_exposed_ports = set()
+
+            image_exposed_ports = set((image["ContainerConfig"]["ExposedPorts"] or {}).keys())
+
+            expected_exposed_ports = set(image_exposed_ports)
             for p in (self.exposed_ports or []):
                 expected_exposed_ports.add("/".join(p))
 
-            actually_exposed_ports = set((container["Config"]["ExposedPorts"] or {}).keys())
+            if self.port_bindings:
+                for container_port in self.port_bindings.keys():
+                    if isinstance(container_port, int):
+                        container_port = "{}/tcp".format(container_port)
+
+                    expected_exposed_ports.add(container_port)
```

Comment:
The problem with including port bindings in the exposed ports section here is that the reload_reason will be incorrect if you change a port binding. You'll never have a chance to reach the actual `# PORT BINDINGS` section later in the function; it'll always short-circuit here at `exposed_ports` if the port bindings have changed.

As much as possible, I tried to keep a one-to-one correspondence between the sections in `get_differing_containers` and the actual settings that users set on the module. Yes, ports that are bound at container start time are also "exposed"; but the module parameter that's used to configure them is `ports:`, not `expose:`.